### ISSUE TITLE
CompareMSExcelFiles Plugin Settings not showing anythings

### ISSUE THREAD

**Issue by fanyan0463** [WinMerge/winmerge#2369]
I am using version 2.16.40.0 X64. However, any other version also encountered same issue. I used open, browse to add .xls file then select CompareMSExcelFiles. However, whenever i click on "Plugin Settings" on CompareMSExcelFiles, there is nothing pop out. I tried it on VMware and it was working. Tired install and uninstall few times but no luck.  Please advise on anything i can try.
<img width="365" alt="winmerge_plugin" src="https://github.com/WinMerge/winmerge/assets/174703584/67e7a800-e3b6-4b35-9e22-944c476caa55">

**Comment by sdottaka** [comment]
The CompareMSExcelFiles Settings dialog is displayed using an outdated component called mshta.exe.

mshta.exe is located in the following paths, but do these files exist?
`C:\Windows\System32\mshta.exe`
`C:\Windows\SysWOW64\mshta.exe`

Also, is there a log in the history of your security software that this program was prevented from starting?

Also, when you press the Plugin Settings button, does any additional icon appear on the taskbar? If it does, it may be that mshta.exe is running but is not visible because it is displayed on an invisible monitor.

**Comment by fanyan0463** [comment]
Hi. mshta.exe existed in both folder since i installed 32 bit and 64 bit before. how to check the log? i do not remember if the program was not able to start. No additional icon appeared, it just nothing has happened.

**Comment by sdottaka** [comment]
How you view logs depends on the security software you are using. Alternatively, if it's logged in the Event Log, you may find crashes of mshta.exe under the Application category in Event Viewer (c:\windows\system32\eventvwr.exe).

If you have time to troubleshoot whether there is indeed an issue with mshta.exe startup, please do the following:

Extract the following zip file:
https://github.com/WinMerge/winmerge/releases/download/v2.16.40/winmerge-2.16.40-x64-exe.zip

Copy the CompareMSExcelFiles.sct file from the attached CompareMSExcelFiles.zip into the extracted WinMerge\MergePlugins folder. Then launch WinMerge, open the Settings dialog for CompareMSExcelFiles, and observe how the message box appears.

[CompareMSExcelFiles.zip](https://github.com/user-attachments/files/16103221/CompareMSExcelFiles.zip)

**Comment by fanyan0463** [comment]
Hi. i do not find mshta crashes under eventvwr.exe. 

I have tried copying [CompareMSExcelFiles.zip] but also not working. By the way, i think all other plugin also encounter same issue. not only with this.

**Comment by sdottaka** [comment]
That's very strange...

- The CompareMSExcelFiles plugin just doesn't show the Settings dialog, but is it possible to extract and compare files?
- Could you please extract the attached testplugin.zip, double-click PluginTest.bat, and let me know the results? In my environment, it appears as shown in the attached image.

[testplugin.zip](https://github.com/user-attachments/files/16109110/testplugin.zip)

![image](https://github.com/WinMerge/winmerge/assets/98126/719c0e82-9d8e-4514-9246-7c0aff924d17)

**Comment by fanyan0463** [comment]
Yes. I am able to compare. but different type of outcome appeared to me. When i used it on vmware, it compared every of my VBA modules. but on this, it only compare the excel sheet. 

Upon trying your batch files, as below:
![image](https://github.com/WinMerge/winmerge/assets/174703584/7b5afe59-b375-409b-8c76-5b39d689c8ff)

**Comment by sdottaka** [comment]
Thank you for testing.
At least, we found that the process has progressed to the point of executing mshta.exe.
Next, could you please extract the attached file testhta.zip and double-click test0bytes.hta to see if a window appears?
Also, could you run test.bat to see if a window appears?
Perhaps the security software installed on your PC deletes the extracted .hta file.
If so, the security software is the cause of the problem.

[testhta.zip](https://github.com/user-attachments/files/16110642/testhta.zip)

**Comment by sdottaka** [comment]
To extract VBA code to text, you must check the Trust access to the VBA project object model check box in the Excel Options window, Trust Center Settings → Macro Settings.

https://support.microsoft.com/zh-tw/topic/programmatic-access-to-office-vba-project-is-denied-960d5265-6592-9400-31bc-b2ddfb94b445
